File: 0001-two-sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 let index1;
 let index2;
var twoSum = function(nums, target) {
    for(let i=0;i<nums.length;i++){
        index1=i;
        for(let j=i+1;j<nums.length;j++){
            if(nums[i]+nums[j]===target){
                index2=j;
                return [index1,index2];
            }

        }
    }
    
};

File: 0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        left = 0 
        right = len(nums) -1

        sorted_nums = [[val,index] for index, val in enumerate(nums)]
        sorted_nums.sort()
        while left < right:
            curr_sum = sorted_nums[left][0] + sorted_nums[right][0]

            if curr_sum == target:
                return [sorted_nums[left][1],sorted_nums[right][1]]
            if curr_sum < target:
                left += 1
            else:
                right -= 1

        
